### PR TITLE
New command added: file-dbcheck: This command checks that all files i…

### PR DESCRIPTION
**Pull request by dariogs**
…n moodledata are on database too.

Output: List with files present on disk but not in database.

### CODE REVIEW

File: Moosh/Command/Moodle23/File/DBCheck.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            }
+            
+            if (!$this->quiet) echo '.';
+        }
+    }
+    
+    private function checkFile($file) {
+	global $DB;
+	
+	$sql = "SELECT id
```
Comment (dkns) [inline Moosh/Command/Moodle23/File/DBCheck.php]: This isn't used anywhere from what I see. Can be safely removed.

File: Moosh/Command/Moodle23/File/DBCheck.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                FROM {files} f
+                WHERE f.contenthash = ?";
+                
+//         if ($filerecord = $DB->get_record_sql($sql, array($file))) {
```
Comment (dkns) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Remove commented out code.

File: Moosh/Command/Moodle23/File/DBCheck.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    if ($DB->count_records("files", array("contenthash"=>$file))!=0) {
```
Comment (dkns) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Indentation.

File: Moosh/Command/Moodle23/File/DBCheck.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        $errors = array();
+        $this->checkDirectory($CFG->dataroot.DIRECTORY_SEPARATOR.'filedir', $errors);
+        if (!$this->quiet) echo "Done.\n\n";
+        
+        if (empty($errors)) {
+            if (!$this->quiet) echo "The contents of moodledata appear to be OK.\n";
+        }
+        
+        else {
```
Comment (dkns) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Indentation/formatting. This should be on line 83.

File: Moosh/Command/Moodle23/File/DBCheck.php [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+	    echo "List of files on disk but not in DB:\n";
```
Comment (dkns) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Indentation

File: Moosh/Command/Moodle23/File/DBCheck.php [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        // Exclude .  ..  warning.txt
+        $list = array_diff($list, array('.', '..', 'warning.txt'));
+        
+        foreach ($list as &$entry) {
+            if (is_dir($dir.$entry)) {
+                // Recursive call on directories
+                $this->checkDirectory($dir.$entry, $errors);
+            }
+            
+            else {
```
Comment (dkns) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Indentation/formatting. This should be on line 43.

File: Moosh/Command/Moodle23/Course/CourseBackup.php [hunk 7]
```diff
@@ -39,10 +40,16 @@ public function execute()
         $shortname = str_replace(' ', '_', $course->shortname);
 
         $options = $this->expandedOptions;
+
+	$cwd=$this->cwd;
```
Comment (dkns) [inline Moosh/Command/Moodle23/Course/CourseBackup.php]: Indentation

**Comment by dkns** [comment]
Hi, thanks a lot for contributing! Could you please fix the few things I have pointed out before we merge this in?

File: Moosh/Command/Moodle23/File/DBCheck.php (hunk 1 above)
Comment (dariogs) [inline Moosh/Command/Moodle23/File/DBCheck.php]: checkFile function is used on line 46. This function test a file hash is present into database.

File: Moosh/Command/Moodle23/Course/CourseBackup.php (hunk 7 above)
Comment (dariogs) [inline Moosh/Command/Moodle23/Course/CourseBackup.php]: Changed

File: Moosh/Command/Moodle23/File/DBCheck.php (hunk 6 above)
Comment (dariogs) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Changed

File: Moosh/Command/Moodle23/File/DBCheck.php (hunk 2 above)
Comment (dariogs) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Removed

File: Moosh/Command/Moodle23/File/DBCheck.php (hunk 3 above)
Comment (dariogs) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Changed

File: Moosh/Command/Moodle23/File/DBCheck.php (hunk 4 above)
Comment (dariogs) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Changed

File: Moosh/Command/Moodle23/File/DBCheck.php (hunk 5 above)
Comment (dariogs) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Duplicated indentation

**Comment by dariogs** [comment]
Hi! I think all problems are solved! Sorry!

File: Moosh/Command/Moodle23/File/DBCheck.php [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+            }
+            
+            if (!$this->quiet) echo '.';
+        }
+    }
+    
+    private function checkFile($file) {
+        global $DB;
+
+        $sql = "SELECT id
```
Comment (dkns) [inline Moosh/Command/Moodle23/File/DBCheck.php]: Where is this used? I can only find one usage of $sql variable and it's this declaration.

**Comment by dkns** [comment]
Thanks a lot!

**Pull request merged**